homeassistant/components/foscam/camera.py

```diff
@@ -1,35 +1,41 @@
 """This component provides basic support for Foscam IP cameras."""
 import asyncio
-import logging
 
 from libpyfoscam import FoscamCamera
 import voluptuous as vol
 
 from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
+from homeassistant.config_entries import SOURCE_IMPORT
 from homeassistant.const import (
     ATTR_ENTITY_ID,
+    CONF_HOST,
     CONF_NAME,
     CONF_PASSWORD,
     CONF_PORT,
     CONF_USERNAME,
 )
 from homeassistant.helpers import config_validation as cv, entity_platform
 
-from .const import DATA as FOSCAM_DATA, ENTITIES as FOSCAM_ENTITIES
+from .const import CONF_STREAM, DOMAIN, LOGGER, SERVICE_PTZ
 
-_LOGGER = logging.getLogger(__name__)
-
-CONF_IP = "ip"
-CONF_RTSP_PORT = "rtsp_port"
-
-DEFAULT_NAME = "Foscam Camera"
-DEFAULT_PORT = 88
-
-SERVICE_PTZ = "ptz"
-ATTR_MOVEMENT = "movement"
-ATTR_TRAVELTIME = "travel_time"
-
-DEFAULT_TRAVELTIME = 0.125
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated("ip"),
+    cv.deprecated(CONF_PASSWORD),
+    cv.deprecated(CONF_USERNAME),
+    cv.deprecated(CONF_NAME),
+    cv.deprecated(CONF_PORT),
+    cv.deprecated("rtsp_port"),
```

Comment:
No need to add all the deprecated flags since we already print a warning in async_setup_platform.